zerver/tests/test_message_edit.py

```diff
@@ -1708,6 +1710,140 @@ def notify(user_id: int) -> Dict[str, Any]:
+        users_notified_via_user_topic_event: List[int] = []
+        for call_args in mock_send_event.call_args_list:
+            (arg_realm, arg_event, arg_notified_users) = call_args[0]
+            if arg_event["type"] == "user_topic":
+                users_notified_via_user_topic_event.append(*arg_notified_users)
+            elif arg_event["type"] == "muted_topics":
+                users_notified_via_muted_topics_event.append(*arg_notified_users)
+        self.assertEqual(
+            sorted(users_notified_via_muted_topics_event),
+            sorted(users_to_be_notified_via_muted_topics_event),
+        )
+        self.assertEqual(
+            sorted(users_notified_via_user_topic_event),
+            sorted(users_to_be_notified_via_user_topic_event),
+        )
+
+        self.assertFalse(
+            topic_has_visibility_policy(
+                hamlet, stream.id, "Topic1", UserTopic.VisibilityPolicy.UNMUTED
+            )
+        )
+        self.assertFalse(
+            topic_has_visibility_policy(
+                cordelia, stream.id, "Topic1", UserTopic.VisibilityPolicy.MUTED
+            )
+        )
+        self.assertFalse(
+            topic_has_visibility_policy(
+                othello, stream.id, "Topic1", UserTopic.VisibilityPolicy.UNMUTED
+            )
+        )
+        self.assertFalse(
+            topic_has_visibility_policy(
+                aaron, stream.id, "Topic1", UserTopic.VisibilityPolicy.UNMUTED
+            )
+        )
+        self.assertTrue(
+            topic_has_visibility_policy(
+                hamlet, stream.id, "Topic2", UserTopic.VisibilityPolicy.UNMUTED
+            )
+        )
+        self.assertTrue(
+            topic_has_visibility_policy(
+                cordelia, stream.id, "Topic2", UserTopic.VisibilityPolicy.MUTED
+            )
+        )
+        self.assertTrue(
+            topic_has_visibility_policy(
+                othello, stream.id, "Topic2", UserTopic.VisibilityPolicy.UNMUTED
+            )
+        )
+        self.assertFalse(
+            topic_has_visibility_policy(
+                aaron, stream.id, "Topic2", UserTopic.VisibilityPolicy.UNMUTED
+            )
+        )
+        self.assertTrue(
+            topic_has_visibility_policy(
+                hamlet, stream.id, change_all_topic_name, UserTopic.VisibilityPolicy.UNMUTED
+            )
+        )
+        self.assertTrue(
+            topic_has_visibility_policy(
+                cordelia, stream.id, change_all_topic_name, UserTopic.VisibilityPolicy.MUTED
+            )
+        )
+        self.assertTrue(
+            topic_has_visibility_policy(
+                othello, stream.id, change_all_topic_name, UserTopic.VisibilityPolicy.UNMUTED
+            )
+        )
+        self.assertFalse(
+            topic_has_visibility_policy(
+                aaron, stream.id, change_all_topic_name, UserTopic.VisibilityPolicy.MUTED
+            )
+        )
```

Comment:
This series of assertions feels pretty hard to read; I wonder if there's a way we could add a test helper that let us compress this a bit more.